server/controllers/search.js

```diff
@@ -0,0 +1,95 @@
+import { Users, Roles, Documents } from '../models';
+import { returnValidationErrors, catchError } from '../helpers/utils';
+
+/**
+ * @description searches for a user
+ * @function searchUser
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const searchUser = (req, res) => {
+  req.checkQuery('q', 'an email or name is required').notEmpty();
+  returnValidationErrors(req, res);
+  const queryString = (req.query.q).toString();
+
+  Users.findAll({
+    where: {
+      $or: [
+        {
+          email: { $iLike: `%${queryString}%` }
+        },
+        {
+          fullname: { $iLike: `%${queryString}%` }
+        }
+      ]
+    },
+    include: [
+      {
+        model: Roles,
+        attributes: { exclude: ['createdAt', 'updatedAt'] },
+      }
+    ],
+    attributes: { exclude: ['password', 'updatedAt'] },
+  })
+  .then((users) => {
+    if (users.length === 0) {
+      return res.status(200).send({
+        status: 'ok',
+        message: 'no user found for your query',
+      });
+    }
+    return res.status(200).send({
+      status: 'ok',
+      users,
+    });
+  })
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description searches for a document using title
+ * @function searchUser
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object
+ */
+const searchDocument = (req, res) => {
+  req.checkQuery('q', 'a document title is required').notEmpty();
+
+  returnValidationErrors(req, res);
+  const queryString = (req.query.q).toString();
```

Comment:
Is there a need for an explicit call to `toString`? Does the value not come as a string by default?